File: src/main/java/com/paragon/client/systems/module/hud/impl/CombatInfo.java
package com.paragon.client.systems.module.hud.impl;

import com.paragon.api.util.render.font.FontUtil;
import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.impl.client.Colours;
import com.paragon.client.systems.module.impl.combat.Aura;
import com.paragon.client.systems.module.impl.combat.AutoCrystal;
import com.paragon.client.systems.module.impl.combat.AutoCrystalRewrite;
import net.minecraft.util.text.TextFormatting;

public class CombatInfo extends HUDModule {

    public static CombatInfo INSTANCE;

    public CombatInfo() {
        super("CombatInfo", "Shows what combat modules are enabled");

        INSTANCE = this;
    }

    @Override
    public void render() {
        FontUtil.drawStringWithShadow("KA " + (Aura.INSTANCE.isEnabled() ? TextFormatting.GREEN + "Enabled" : TextFormatting.RED + "Disabled"), getX(), getY(), Colours.mainColour.getValue().getRGB());
        FontUtil.drawStringWithShadow("CA " + (AutoCrystal.INSTANCE.isEnabled() ? TextFormatting.GREEN + "Enabled" : TextFormatting.RED + "Disabled"), getX(), getY() + FontUtil.getHeight(), Colours.mainColour.getValue().getRGB());
        FontUtil.drawStringWithShadow("ACR " + (AutoCrystalRewrite.INSTANCE.isEnabled() ? TextFormatting.GREEN + "Enabled" : TextFormatting.RED + "Disabled"), getX(), getY() + FontUtil.getHeight() * 2, Colours.mainColour.getValue().getRGB());
    }

    @Override
    public float getWidth() {
        // Longest
        return FontUtil.getStringWidth("ACR Disabled");
    }

    @Override
    public float getHeight() {
        return FontUtil.getHeight() * 3;
    }

}

File: src/main/kotlin/com/paragon/client/systems/module/hud/impl/CombatInfo.kt
package com.paragon.client.systems.module.hud.impl

import com.paragon.api.util.render.font.FontUtil
import com.paragon.api.util.render.font.FontUtil.drawStringWithShadow
import com.paragon.api.util.render.font.FontUtil.getStringWidth
import com.paragon.client.systems.module.hud.HUDModule
import com.paragon.client.systems.module.impl.client.Colours
import com.paragon.client.systems.module.impl.combat.Aura
import com.paragon.client.systems.module.impl.combat.AutoCrystal
import net.minecraft.util.text.TextFormatting

/**
 * @author Surge
 */
object CombatInfo : HUDModule("CombatInfo", "Shows what combat modules are enabled") {

    override fun render() {
        drawStringWithShadow(
            "KA " + if (Aura.isEnabled) TextFormatting.GREEN.toString() + "Enabled" else TextFormatting.RED.toString() + "Disabled",
            x,
            y,
            Colours.mainColour.value.rgb
        )

        drawStringWithShadow(
            "CA " + if (AutoCrystal.isEnabled) TextFormatting.GREEN.toString() + "Enabled" else TextFormatting.RED.toString() + "Disabled",
            x,
            y + FontUtil.getHeight(),
            Colours.mainColour.value.rgb
        )
    }

    override var width = 10F
        get() = getStringWidth("AC Disabled")

    override var height = 20F
        get() = FontUtil.getHeight() * 3

}